include/ginkgo/core/base/lin_op.hpp

```diff
@@ -779,6 +779,58 @@ class EnableAbsoluteComputation : public AbsoluteComputable {
 };
 
 
+/**
+ * Adds the operation M <- a I + b M for matrix M, identity
+ * operator I and scalars a and b, where M is the calling object.
+ */
+class ScaledIdentityAddable {
+public:
+    /**
+     * Scales this and adds another scalar times the identity to it.
+     *
+     * @param a  Scalar to multiply the identity operator by before adding.
+     * @param b  Scalar to multiply this before adding the scaled identity to
+     *   it.
+     */
+    virtual void add_scaled_identity(const LinOp* const a,
+                                     const LinOp* const b) = 0;
+};
+
+
+/**
+ * Implements common checks for the operation M <- a I + b M for matrix M,
+ * identity operator I and scalars a and b, where M is the calling object.
+ *
+ * @tparam ConcreteType  The concrete type that should have this operation
+ *                       (the CRTP parameter).
+ */
+template <typename ConcreteType>
+class EnableScaledIdentityAddition : public ScaledIdentityAddable {
+public:
+    /**
+     * Scales this and adds another scalar times the identity to it.
+     *
+     * @param a  Scalar to multiply the identity operator by before adding.
+     * @param b  Scalar to multiply this before adding the scaled identity to
+     *   it.
+     */
+    void add_scaled_identity(const LinOp* const a,
```

Comment:
Not a blocker or anything, just a comment: I can't exactly say why, but this parameter order seems the wrong way round for me.